daemon/qrexec-daemon.c

```diff
@@ -1274,11 +1309,13 @@ static void handle_message_from_agent(void)
 
             /* sanitize start */
             ENSURE_NULL_TERMINATED(untrusted_params3.target_domain);
-            ENSURE_NULL_TERMINATED(untrusted_params3.request_id.ident);
             untrusted_service_name[service_name_len-1] = 0;
             sanitize_name(untrusted_params3.target_domain, "@:");
-            sanitize_name(untrusted_params3.request_id.ident, " ");
             sanitize_name(untrusted_service_name, "+");
+            if (!validate_request_id(&untrusted_params3.request_id, "MSG_TRIGGER_SERVICE3")) {
+                send_service_refused(vchan, &untrusted_params3.request_id);
```

Comment:
This leaks `untrusted_service_name` (issue detected by the tests). I'll fix that with commit on top.